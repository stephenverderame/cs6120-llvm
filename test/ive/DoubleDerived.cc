// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 10;
    for (int i = 0; i < 100; ++i) {
        int k = i * m;
        printf("%d\n", k);
        int p = k + 10;
        printf("%d\n", p);
        int j = k + 20;
        printf("%d\n", j);
    }
}
// NOLINTEND