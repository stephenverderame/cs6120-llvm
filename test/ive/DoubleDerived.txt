0
10
20
10
20
30
20
30
40
30
40
50
40
50
60
50
60
70
60
70
80
70
80
90
80
90
100
90
100
110
100
110
120
110
120
130
120
130
140
130
140
150
140
150
160
150
160
170
160
170
180
170
180
190
180
190
200
190
200
210
200
210
220
210
220
230
220
230
240
230
240
250
240
250
260
250
260
270
260
270
280
270
280
290
280
290
300
290
300
310
300
310
320
310
320
330
320
330
340
330
340
350
340
350
360
350
360
370
360
370
380
370
380
390
380
390
400
390
400
410
400
410
420
410
420
430
420
430
440
430
440
450
440
450
460
450
460
470
460
470
480
470
480
490
480
490
500
490
500
510
500
510
520
510
520
530
520
530
540
530
540
550
540
550
560
550
560
570
560
570
580
570
580
590
580
590
600
590
600
610
600
610
620
610
620
630
620
630
640
630
640
650
640
650
660
650
660
670
660
670
680
670
680
690
680
690
700
690
700
710
700
710
720
710
720
730
720
730
740
730
740
750
740
750
760
750
760
770
760
770
780
770
780
790
780
790
800
790
800
810
800
810
820
810
820
830
820
830
840
830
840
850
840
850
860
850
860
870
860
870
880
870
880
890
880
890
900
890
900
910
900
910
920
910
920
930
920
930
940
930
940
950
940
950
960
950
960
970
960
970
980
970
980
990
980
990
1000
990
1000
1010
