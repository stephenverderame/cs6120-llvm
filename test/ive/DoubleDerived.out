Loop: 
1:                                                ; preds = %11, %0
  %.0 = phi i32 [ 0, %0 ], [ %12, %11 ]
  %2 = icmp slt i32 %.0, 100
  br i1 %2, label %4, label %3

Basic IV:   %.0 = phi i32 [ 0, %0 ], [ %12, %11 ]
Basic IV:   %12 = add nsw i32 %.0, 1
Derived IV: %5 = %.0 * 10
Derived IV: %7 = %.0 * 10 + 10
Derived IV: %9 = %.0 * 10 + 20
